lib/matplotlib/axes/_axes.py

```diff
@@ -1960,6 +1965,11 @@ def make_iterable(x):
                 left *= nbars
             if len(height) == 1:
                 height *= nbars
+            if nbars and cbook.is_sequence_of_strings(bottom):
+                ticks_loc = [i + height[i]/2 for i in range(nbars)]
```

Comment:
I would do this as `[i + h / 2 for i, h in enumerate(height)]`, same for the logic in `bar`